reana_workflow_controller/tasks.py

```diff
@@ -99,3 +105,24 @@ def _update_job_progress(workflow_uuid, msg):
                     run_job.run_id = current_run.id_
                     run_job.job_id = job_id
                     Session.add(run_job)
+
+
+def _update_job_cache(msg):
+    """Update caching information for finished job."""
+    cmd = msg['caching_info']['job_spec']['cmd']
+    # removes cd to workspace, to be refactored
+    clean_cmd = cmd.split(';')[1]
+    msg['caching_info']['job_spec']['cmd'] = clean_cmd
+    input_hash = calculate_job_input_hash(msg['caching_info']['job_spec'],
+                                          msg['caching_info']['workflow_json'])
+    directory_hash = calculate_hash_of_dir(
```

Comment:
As spotted today, RJC controller calculates the hash before the job has been started (see [here](https://github.com/reanahub/reana-job-controller/pull/74/files#diff-8ee72e3f835364b2cb6e08b399497ce7R98)) but for steps that do thinks like `- echo 'Great it is Friday' > test.txt` which are fast to execute, the workflow workspace will change by the moment this [line](https://github.com/reanahub/reana-workflow-controller/pull/104/files#diff-d95dcaa8d671d2a3c798faa7fd757a3fR118) is executed, calculating a different `workspace_hash`